components/tools/OmeroWeb/omeroweb/webclient/views.py

```diff
@@ -605,43 +605,55 @@ def api_container_list(request, conn=None, **kwargs):
     # parents), screens and plates (without parents). This is fine for
     # the first page, but the second page may not be what is expected.
 
+    r = dict()
     try:
         # Get the projects
-        projects = tree.marshal_projects(conn=conn,
-                                         group_id=group_id,
-                                         experimenter_id=experimenter_id,
-                                         page=page,
-                                         limit=limit)
+        r['projects'] = tree.marshal_projects(
+            conn=conn,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
 
         # Get the orphaned datasets (without project parents)
-        datasets = tree.marshal_datasets(conn=conn,
-                                         orphaned=True,
-                                         group_id=group_id,
-                                         experimenter_id=experimenter_id,
-                                         page=page,
-                                         limit=limit)
+        r['datasets'] = tree.marshal_datasets(
+            conn=conn,
+            orphaned=True,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
 
         # Get the screens for the current user
-        screens = tree.marshal_screens(conn=conn,
-                                       group_id=group_id,
-                                       experimenter_id=experimenter_id,
-                                       page=page,
-                                       limit=limit)
+        r['screens'] = tree.marshal_screens(
+            conn=conn,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
 
         # Get the orphaned plates (without project parents)
-        plates = tree.marshal_plates(conn=conn,
-                                     orphaned=True,
-                                     group_id=group_id,
-                                     experimenter_id=experimenter_id,
-                                     page=page,
-                                     limit=limit)
-
+        r['plates'] = tree.marshal_plates(
+            conn=conn,
+            orphaned=True,
+            group_id=group_id,
+            experimenter_id=experimenter_id,
+            page=page,
+            limit=limit)
         # Get the orphaned images container
-        orphaned = tree.marshal_orphaned(conn=conn,
-                                         group_id=group_id,
-                                         experimenter_id=experimenter_id,
-                                         page=page,
-                                         limit=limit)
+        if (conn.isAdmin() or conn.isLeader() or
```

Comment:
```conn.isLeader()``` tells you if you are leader of your default group, not necessarily the group you're currently looking at. I think you need to use ```conn.isLeader( gid=request.session.get('active_group') )```